servant-server/test/Servant/ServerSpec.hs

```diff
@@ -339,6 +339,15 @@ qpServer = queryParamServer :<|> qpNames :<|> qpCapitalize :<|> qpAge :<|> qpAge
         queryParamServer (Just name_) = return alice{name = name_}
         queryParamServer Nothing = return alice
 
+-- test query parameters rewriter, commonly implemented using WAI Middleware
+sampleRewrite :: Query -> Query
+sampleRewrite = fmap mapper where 
```

Comment:
This should be a `Middleware`, `serve queryParamApi qpServer` produces an `Application`, to which it could be applied.

I'll do the change myself